src/graphql/mirror.js

```diff
@@ -318,6 +318,18 @@ export class Mirror {
           CREATE INDEX idx_connection_entries__connection_id
           ON connection_entries (connection_id)
         `,
+        dedent`\
+          CREATE TABLE network_log (
+              rowid INTEGER PRIMARY KEY,
+              query TEXT NOT NULL,
+              query_parameters TEXT NOT NULL,
+              request_timestamp INTEGER NOT NULL,
+              response TEXT,
+              response_timestamp INTEGER,
+              update_id INTEGER,
+              FOREIGN KEY(update_id) REFERENCES updates(rowid)
```

Comment:
OK: You could add `CHECK` constraints here that enforce that the
relevant groups of parameters are all null or not null together (e.g.,
`response` and `response_timestamp` should be either both absent or both
present), but it’d also be fine to leave them out.

(IMHO, the underlying ontology here is that there are three relations:
say, `request_log`, `response_log`, and `update_log`, linked by foreign
keys, and within each relation all columns are non-null. Using this
compact single-relation form is fine because it’s probably more
convenient for interactive use, which is a primary design constraint.)

Reply:
updated

Reply:
Updated. I had trouble testing this however, I believe b/c of the `+` attached to the `timestamp` parameter passed to `run`. Passing `null` to `_logRequest` for the `timestamp` parameter didn't trigger a constraint violation, but passing `null` directly to the `run` function in place of `+timestamp` did.